lib/mixlib/install/backend/artifactory.rb

```diff
@@ -0,0 +1,84 @@
+#
+# Author:: Patrick Wright (<patrick@chef.io>)
+# Copyright:: Copyright (c) 2015 Chef, Inc.
+# License:: Apache License, Version 2.0
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+#
+
+require "net/http"
+require "json"
+require "mixlib/install/artifact_info"
+require "artifactory"
+
+module Mixlib
+  class Install
+    class Backend
+      class Artifactory
+        ARTIFACTORY_ENDPOINT = "http://artifactory.chef.co".freeze
+
+        attr_reader :options
+        attr_reader :client
+
+        def initialize(options)
+          @options = options
+          @client = ::Artifactory::Client.new(endpoint: ARTIFACTORY_ENDPOINT)
+        end
+
+        def info
+          params = {
+            "repos" => "omnibus-current-local",
+            "omnibus.version" => options.product_version
+          }
+
+          if options.platform
```

Comment:
I think raising an error when only `p` is provided will be better. That way we will not get usage we do not want to support moving forward :smile: